test/parallel/test-stream-typedarrays.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const { Readable, Writable } = require('stream');
+
+const messageBuffer = Buffer.from('ABCDEFGH'); // Needs to be a multiple of 8
+const toBeWritten = [...common.getArrayBufferViews(messageBuffer)];
+const nthWrittenBuffer = (n) => Buffer.from(
+  toBeWritten[n].buffer,
+  toBeWritten[n].byteOffset,
+  toBeWritten[n].byteLength
+);
+
+{
+  // Simple Writable test.
+
+  let n = 0;
+  const writable = new Writable({
+    write: common.mustCall((chunk, encoding, cb) => {
+      assert(!(chunk instanceof Buffer));
+      assert(ArrayBuffer.isView(chunk));
+      assert.deepStrictEqual(chunk, nthWrittenBuffer(n++));
```

Comment:
Instances of `Buffer` could not be `deepStrictEqual ` to those of `TypedArray`, could they?

Maybe try:
```js
assert.deepStrictEqual(Buffer.from(chunk.buffer, chunk.byteOffset, chunk.length), nthWrittenBuffer(n++));
```